android/src/main/java/com/tailscale/ipn/ShareActivity.kt

```diff
@@ -92,7 +94,20 @@ class ShareActivity : ComponentActivity() {
             val nameCol = c.getColumnIndex(OpenableColumns.DISPLAY_NAME)
             val sizeCol = c.getColumnIndex(OpenableColumns.SIZE)
             c.moveToFirst()
-            val name = c.getString(nameCol)
+            val name: String =
+                c.getString(nameCol)
+                    ?: run {
+                      // For some reason, some content resolvers don't return a name.
+                      // Try to build a name from a random integer plus file extension
+                      // (if type can be determined), else just a random integer.
```

Comment:
Maybe there's a better algorithm we can use than just some random integer, I'm not sure.